src/coordinates.rs

```diff
@@ -89,28 +89,24 @@ impl_from_integer!(u16);
 
 impl CoordinateNumber {
     pub fn gerber(&self, format: &CoordinateFormat) -> Result<String, GerberError> {
-        // Format invariants
-        if format.decimal > DECIMAL_PLACES_CHARS {
-            return Err(GerberError::CoordinateFormatError("Invalid precision: Too high!".into()))
+        fn rounded_div(dividend: i64, divisor: i64) -> i64 {
+            if dividend.is_positive() == divisor.is_positive() {
+                (dividend + (divisor / 2)) / divisor
+            } else {
+                (dividend - (divisor / 2)) / divisor
+            }
         }
 
-        // If value is 0, return corresponding string
-        if self.nano == 0 {
-            return Ok("0".to_string());
+        if format.decimal > DECIMAL_PLACES_CHARS {
+            return Err(GerberError::CoordinateFormatError("Invalid precision: Too high!".into()))
         }
-
-        // Get integer part by doing floor division
-        let integer: i64 = self.nano / DECIMAL_PLACES_FACTOR;
-        if integer > 10i64.pow(format.integer as u32) {
-            return Err(GerberError::CoordinateFormatError("Decimal is too large for chosen format".into()));
+        if self.nano >= 10_i64.pow((format.integer + DECIMAL_PLACES_CHARS) as u32) {
+            return Err(GerberError::CoordinateFormatError("Number is too large for chosen format!".into()));
         }
 
-        // Get decimal part with proper rounding
-        let divisor: i64 = 10i64.pow((DECIMAL_PLACES_CHARS - format.decimal) as u32);
-        let decimal: i64 = ((self.nano % DECIMAL_PLACES_FACTOR).abs() + (divisor / 2)) / divisor;
-
-        // Convert to string
-        Ok(format!("{}{:0<width$}", integer, decimal, width=format.decimal as usize))
+        let divisor: i64 = 10_i64.pow((DECIMAL_PLACES_CHARS - format.decimal) as u32);
```

Comment:
I'm not actually sure if we need the `i64` suffix at all. Could you try without it (also on line 103)? Maybe it can be derived.